node/src/components/consensus/era_supervisor.rs

```diff
@@ -660,15 +694,15 @@ where
 
             let slashed = self
                 .era_supervisor
-                .iter_past_other(era_id, self.era_supervisor.bonded_eras)
+                .iter_past(era_id, self.era_supervisor.bonded_eras)
                 .filter_map(|old_id| switch_blocks.get(&old_id).and_then(|bhdr| bhdr.era_end()))
                 .flat_map(|era_end| era_end.equivocators.clone())
                 .collect();
 
             let booking_era_id = EraId(
                 era_id
                     .0
-                    .saturating_sub(self.era_supervisor.protocol_config.auction_delay + 1),
+                    .saturating_sub(self.era_supervisor.protocol_config.auction_delay),
```

Comment:
So this is now the era whose key block is the booking block for `era_id`, and that's the reason we don't need to subtract `1` anymore? :dizzy_face:

Reply:
Ok, I probably made this unnecessarily confusing :sweat_smile: 

The reason we don't need to subtract 1 here anymore is that the `switch_blocks` map is now `era_id → switch block of era era_id-1` - so for each era ID it contains the switch block used to _initialize_ that era, not the final block of that era.